rs/pocket_ic_server/src/pocket_ic.rs

```diff
@@ -2077,6 +2081,92 @@ impl PocketIcSubnets {
             "Failed to complete execution of method {method} on canister {canister_id} after 100 rounds."
         );
     }
+
+    fn sync_registry_from_canister(&mut self) {
```

Comment:
The approach here works fine.  It seems like nns_data_provider library does a good deal of the low level grunt work in this method (getting the inlined_deltas), so I think you could potentially simplify this if you want to, and make this code require fewer changes in the future if something changes in Registry.

Reply:
LG